src/context/AppProvider/hooks/useAccountsFetchQuery.tsx

```diff
@@ -0,0 +1,205 @@
+import { usePrevious } from '@chakra-ui/react'
+import type { AccountId, ChainId } from '@shapeshiftoss/caip'
+import { fromAccountId } from '@shapeshiftoss/caip'
+import { isLedger } from '@shapeshiftoss/hdwallet-ledger'
+import { MetaMaskMultiChainHDWallet } from '@shapeshiftoss/hdwallet-metamask-multichain'
+import type { AccountMetadataById } from '@shapeshiftoss/types'
+import { skipToken, useQuery } from '@tanstack/react-query'
+import { useCallback, useEffect, useMemo } from 'react'
+import { usePlugins } from 'context/PluginProvider/PluginProvider'
+import { useIsSnapInstalled } from 'hooks/useIsSnapInstalled/useIsSnapInstalled'
+import { useWallet } from 'hooks/useWallet/useWallet'
+import { walletSupportsChain } from 'hooks/useWalletSupportsChain/useWalletSupportsChain'
+import { deriveAccountIdsAndMetadata } from 'lib/account/account'
+import { isUtxoChainId } from 'lib/utils/utxo'
+import { portfolio, portfolioApi } from 'state/slices/portfolioSlice/portfolioSlice'
+import { selectEnabledWalletAccountIds } from 'state/slices/selectors'
+import { txHistoryApi } from 'state/slices/txHistorySlice/txHistorySlice'
+import { useAppDispatch, useAppSelector } from 'state/store'
+
+export const useAccountsFetchQuery = () => {
+  const dispatch = useAppDispatch()
+  const enabledWalletAccountIds = useAppSelector(selectEnabledWalletAccountIds)
+  const { supportedChains } = usePlugins()
+  const { isSnapInstalled } = useIsSnapInstalled()
+  const previousIsSnapInstalled = usePrevious(isSnapInstalled)
+  const { deviceId, wallet } = useWallet().state
+
+  const hasManagedAccounts = useMemo(() => {
+    // MM without snap doesn't allow account management - if the user just installed the snap, we know they don't have managed accounts
+    if (!previousIsSnapInstalled && isSnapInstalled) return false
+    // We know snap wasn't just installed in this render - so if there are any requestedAccountIds, we assume the user has managed accounts
+    return enabledWalletAccountIds.length > 0
+  }, [isSnapInstalled, previousIsSnapInstalled, enabledWalletAccountIds.length])
+
+  // Fetch portfolio for all managed accounts as a side-effect if they exist instead of going through the initial account detection flow.
+  // This ensures that we have fresh portfolio data, but accounts added through account management are not accidentally blown away.
+  useEffect(() => {
+    const { getAllTxHistory } = txHistoryApi.endpoints
+
+    // Note no force refetch here - only fetch Tx history once per acccount
+    enabledWalletAccountIds.forEach(accountId => {
+      dispatch(portfolioApi.endpoints.getAccount.initiate({ accountId, upsertOnFetch: true }))
+    })
```

Comment:
The actual fix:

- remove `if (!isAccountManagementEnabled) return` and `if (isFetching) return` checks, meaning this will referentially invalidate and rerun everytime `enabledWalletAccountIds` does, as it should. 
- add `enabledWalletAccountIds.map(requestedAccountId (dispatch(getAllTxHistory.initiate(requestedAccountId)))` in the effect 
There was another similar-looking Tx history call within the query itself which couldn't possibly work, since `enabledWalletAccountIds` will be empty when the query runs (which is "correct" as far as our flow goes)

The bug actually came from https://github.com/shapeshift/web/pull/7932, not https://github.com/shapeshift/web/pull/7917, though since the former was merged into the latter instead of the root of the stack being merged in, it looks like the latter was the culprit diff-wise, and is a good reminder to always merge downmost PRs first, to keep things granular and avoiding having to debug a bigly diff when a small one on top of it was the culprit.